### PR TITLE
Improve test-suite.yml for CI failing when disabling tokenization

### LINKED ISSUE #3991: CI failing when disabling tokenization
We run some tests every day, instead of on every PR
One of the [test is to ensure we can disable the tokenization](https://github.com/meilisearch/meilisearch/blob/29adfc2f6835fb17a584f5e2bcf7d7acde48e44b/.github/workflows/test-suite.yml#L108-L129) (when compiling Meilisearch) without any issue.
Currently, our CI testing this is failing every day (not randomly).

It seems like disabling all the features, i.e. `no-default-features`, of Meilisearch no longer removes the lindera-related libraries from Meilisearch and continues to compile them. The lindera libraries are used by the direct dependency charabia. The main goal of disabling lindera was to reduce the binary size and the compilation time.

You should probably take a look at [the charabia _Cargo.toml_](https://github.com/meilisearch/charabia/blob/de62ab9b889061126b0a8473aa53bf6288a99679/charabia/Cargo.toml#L27C1-L38) and make some of the libraries optional.

I tried to delete all the GitHub Actions cache, but nothing changed, and the CI is still failing
https://github.com/meilisearch/meilisearch/actions/runs/5817159170/job/15771397666

```
  ...
  Downloaded hmac v0.12.1
  Downloaded generic-array v0.14.7
  Downloaded futures-executor v0.3.28
  Downloaded futures-macro v0.3.28
  Downloaded serde_urlencoded v0.7.1
  Downloaded cfg-if v1.0.0
  Downloaded bytes v1.4.0
  Downloaded either v1.8.1
Error: Process completed with exit code 1.
```

The main problem is the error is really vague.

### PR DESCRIPTION
**Pull request by harshau007**
# Pull Request
Added New run command for `Run cargo tree without default features and check lindera is not present`

## Related issue
Related to #3991 

## What does this PR do?
- fixes `CI failing when disabling tokenization` by adding if-then block for `Run cargo tree without default features and check lindera is not present` in `.github/workflows/test-suite.yml`

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

**Comment by harshau007** [comment]
Hey @curquiza, It would be great if you could review my pr and provide your valuable feedback.

**Comment by Kerollmops** [comment]
Hey @harshau007 👋

Would it be possible to display a message when the Lindera library is found and shouldn't instead of the other way around, please? The main goal of this CI is to report when the lindera crate is found but mustn't, as we disabled most of the tokenization feature (with `--no-default-features`).

**Comment by harshau007** [comment]
Hey @Kerollmops, made the changes you asked for CI will display message `lindera is present` only if its installed or else it will continue with further steps

File: .github/workflows/test-suite.yml [hunk 1]
```diff
@@ -123,7 +123,9 @@ jobs:
           override: true
       - name: Run cargo tree without default features and check lindera is not present
         run: |
-          cargo tree -f '{p} {f}' -e normal --no-default-features | grep lindera -vqz
+          if cargo tree -f '{p} {f}' -e normal --no-default-features | grep -vqz lindera; then
+            echo "lindera found"
```
Comment (Kerollmops) [inline .github/workflows/test-suite.yml:127]:
```suggestion
            echo "lindera found has been found in the sources and shouldn't"
```

**Review by Kerollmops — DISMISSED** [review]

**Comment by curquiza** [comment]
bors merge

**Comment by curquiza** [comment]
bors merge-

**Comment by curquiza** [comment]
Merging by hand, looks like we have a crash with bors...

Capture d’écran 2023-09-11 à 12 27 29


I think it's due to the permission when I read this [discussion](https://forum.bors.tech/t/bors-crashing-resource-not-accessible-by-integration/571/7). But we already have write & read permission access on this repo 🤷‍♀️

**Pull request merged**